### ISSUE TITLE
Dimers often failing to converge in SCF when using DFT

### ISSUE THREAD

**Issue by BradenDKelly** [psi4/psi4#2338]
Hello, I have been trying to geometry optimize >100 dimer pairs of organic molecules. I have a 60% success rate. I am struggling with the 40% that fail. I have tried using second order convergence and level shifting. Below is an example of a citric acid dimer that fails to converge in 100 scf iterations. The systems all seem to follow the same trend. The first couple of iterations converge the wavefunction, but eventually the scf fails to converge.

The starting dimer configuration is generated using a steepest descent MM energy minimization.

I can provide the entire log file, or other files as well. On other runs I have tried using LEVEL_SHIFT, but it did not help.

```
molecule dimer {
C   999.407602 1002.751600 1001.828601
C   1000.887601 1002.788600 1001.459601
O   1001.420601 1002.207601 1000.537601
O   1001.632601 1003.490600 1002.306601
C   998.553602 1001.680601 1001.080601
C   997.058603 1001.817601 1001.472601
C   996.802603 1001.523601 1002.938600
O   996.673603 1002.368601 1003.793600
O   996.522603 1000.257601 1003.219600
C   999.058602 1000.222601 1001.385601
O   999.978601 999.942601 1002.133601
O   998.491602 999.266602 1000.641601
O   998.619602 1001.964601 999.673602
H   999.003602 1003.752600 1001.641601
H   999.352602 1002.590600 1002.911600
H   1002.505600 1003.321600 1001.914601
H   996.453603 1001.125601 1000.874601
H   996.697603 1002.829600 1001.259601
H   996.300603 1000.363601 1004.159600
H   999.068602 998.498602 1000.848601
H   999.237602 1001.315601 999.248602
--
C   1000.474601 997.570602 997.565602
C   999.917601 998.148602 996.273603
O   1000.520601 998.883602 995.519603
O   998.741602 997.650602 995.912603
C   1001.785601 998.271602 998.044602
C   1002.427600 997.524602 999.252602
C   1001.663601 997.719602 1000.554601
O   1000.490601 997.439602 1000.697601
O   1002.249601 998.561602 1001.401601
C   1001.507601 999.774602 998.424602
O   1000.400601 1000.275601 998.494602
O   1002.573600 1000.457601 998.849602
O   1002.730600 998.191602 996.966603
H   1000.645601 996.500603 997.405602
H   999.684602 997.657602 998.318602
H   998.630602 998.128602 995.073603
H   1003.450600 997.893602 999.402602
H   1002.505600 996.451603 999.055602
H   1001.477601 998.875602 1001.920601
H   1002.138601 1001.212601 999.307602
H   1002.322601 998.665602 996.205603
units angstrom
}
memory 40 Gb

set {
    basis def2-tzvp
    maxiter 100
    g_convergence gau
    soscf true
    soscf_max_iter 40
    }
set_num_threads(20)
E, wfn = optimize('B3LYP-D3', bsse_type='cp', return_wfn=True)
fchk(wfn, 'citric_acid_bsse_cp.fchk')

def saveFile(name, string):
    f=open(name, 'w')
    f.write(string)
    f.close()

saveFile('citric_acid.optimized.dimer_geometry_bsse_cp.log', wfn.molecule().save_string_xyz_file())
```

Here is the first iteration
```
  ==> Pre-Iterations <==

  SCF Guess: Superposition of Atomic Densities via on-the-fly atomic UHF (no occupation information).

   -------------------------
    Irrep   Nso     Nmo
   -------------------------
     A        451     451
   -------------------------
    Total     451     451
   -------------------------

  ==> Iterations <==

                           Total Energy        Delta E     RMS |[F,P]|

   @DF-RKS iter SAD:  -759.49493403045983   -7.59495e+02   0.00000e+00
   @DF-RKS iter   1:  -759.57044795569880   -7.55139e-02   3.31257e-03 DIIS
   @DF-RKS iter   2:  -758.78767299108517    7.82775e-01   4.58393e-03 DIIS
   @DF-RKS iter   3:  -760.32534769892425   -1.53767e+00   9.55400e-04 SOKS, nmicro=7
   @DF-RKS iter   4:  -760.39557129980335   -7.02236e-02   4.93871e-05 SOKS, nmicro=8
   @DF-RKS iter   5:  -760.39613633741988   -5.65038e-04   7.21946e-07 SOKS, nmicro=8
   @DF-RKS iter   6:  -760.39613642852896   -9.11091e-08   1.76332e-09 SOKS, nmicro=8
   @DF-RKS iter   7:  -760.39613642852896    0.00000e+00   5.70403e-12 SOKS, nmicro=conv
  Energy and wave function converged.
```

and here is the final iteration that fails

```
 ==> Pre-Iterations <==

  SCF Guess: Orbitals guess was supplied from a previous computation.

   -------------------------------------------------------
    Irrep   Nso     Nmo     Nalpha   Nbeta   Ndocc  Nsocc
   -------------------------------------------------------
     A        902     902      50      50      50       0
   -------------------------------------------------------
    Total     902     902      50      50      50       0
   -------------------------------------------------------

  ==> Iterations <==

                           Total Energy        Delta E     RMS |[F,P]|

   @DF-RKS iter   0:  -756.64555164402248   -7.56646e+02   2.23119e-03
   @DF-RKS iter   1:  -756.44709462410424    1.98457e-01   2.61941e-03 DIIS
   @DF-RKS iter   2:  -696.79549781482638    5.96516e+01   8.79521e-03 DIIS
   @DF-RKS iter   3:  -757.70517729196388   -6.09097e+01   2.61678e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter   4:  -743.37798168217728    1.43272e+01   5.03487e-03 DIIS
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter   5:  -756.65675968040807   -1.32788e+01   3.12016e-03 DIIS
   @DF-RKS iter   6:  -757.18710490514206   -5.30345e-01   2.04901e-03 SOKS, nmicro=40
   @DF-RKS iter   7:  -749.08851900767434    8.09859e+00   3.21116e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter   8:  -714.30385026747103    3.47847e+01   8.06060e-03 DIIS
   @DF-RKS iter   9:  -757.30477472616599   -4.30009e+01   2.00054e-03 SOKS, nmicro=40
   @DF-RKS iter  10:  -750.16020340117416    7.14457e+00   3.81913e-03 SOKS, nmicro=40
   @DF-RKS iter  11:  -740.49687504339590    9.66333e+00   4.95648e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  12:  -731.84361006071981    8.65326e+00   5.79602e-03 DIIS
   @DF-RKS iter  13:  -757.61845594897056   -2.57748e+01   1.90863e-03 SOKS, nmicro=40
   @DF-RKS iter  14:  -752.69076511176888    4.92769e+00   2.46542e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  15:  -746.30543656371788    6.38533e+00   3.67561e-03 DIIS
   @DF-RKS iter  16:  -757.09831604310784   -1.07929e+01   2.16538e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  17:  -737.69985235900867    1.93985e+01   5.70342e-03 DIIS
   @DF-RKS iter  18:  -757.51138002491950   -1.98115e+01   1.95654e-03 SOKS, nmicro=40
   @DF-RKS iter  19:  -754.60371905617808    2.90766e+00   2.41872e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  20:  -725.99330440541530    2.86104e+01   6.05707e-03 DIIS
   @DF-RKS iter  21:  -757.42025013784109   -3.14269e+01   1.99422e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  22:  -752.45843332397510    4.96182e+00   2.85671e-03 DIIS
   @DF-RKS iter  23:  -759.31039071524231   -6.85196e+00   1.36021e-03 SOKS, nmicro=40
   @DF-RKS iter  24:  -757.28547942834871    2.02491e+00   2.05134e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  25:  -751.64112255339569    5.64436e+00   3.87070e-03 DIIS
   @DF-RKS iter  26:  -757.19460474836558   -5.55348e+00   2.01715e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  27:  -745.51246724615862    1.16821e+01   4.51674e-03 DIIS
   @DF-RKS iter  28:  -757.12516483532829   -1.16127e+01   2.04494e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  29:  -746.03911568891294    1.10860e+01   3.55967e-03 DIIS
   @DF-RKS iter  30:  -757.66681851063925   -1.16277e+01   1.84247e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  31:  -746.45803660662887    1.12088e+01   3.28905e-03 DIIS
   @DF-RKS iter  32:  -757.56206221286379   -1.11040e+01   1.89236e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  33:  -747.65024063569217    9.91182e+00   3.70186e-03 DIIS
   @DF-RKS iter  34:  -757.51177313836968   -9.86153e+00   1.93518e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  35:  -742.16361953202068    1.53482e+01   4.88585e-03 DIIS
   @DF-RKS iter  36:  -757.41327433562515   -1.52497e+01   1.98821e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  37:  -745.20547489084629    1.22078e+01   3.94490e-03 DIIS
   @DF-RKS iter  38:  -757.51356397289487   -1.23081e+01   1.92618e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  39:  -747.92804630313231    9.58552e+00   3.50103e-03 DIIS
   @DF-RKS iter  40:  -757.55034022435677   -9.62229e+00   1.93180e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  41:  -747.63711597281156    9.91322e+00   4.78663e-03 DIIS
   @DF-RKS iter  42:  -757.40890819341848   -9.77179e+00   1.99481e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  43:  -741.71952296931534    1.56894e+01   5.02152e-03 DIIS
   @DF-RKS iter  44:  -757.45692330245845   -1.57374e+01   1.96809e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  45:  -749.44376442136877    8.01316e+00   3.47838e-03 DIIS
   @DF-RKS iter  46:  -757.56212366630984   -8.11836e+00   1.94273e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  47:  -742.13626778876608    1.54259e+01   5.22242e-03 DIIS
   @DF-RKS iter  48:  -753.25750326994705   -1.11212e+01   3.08251e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  49:  -741.58227350186746    1.16752e+01   3.75047e-03 DIIS
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  50:  -753.51766570383381   -1.19354e+01   3.06550e-03 DIIS
   @DF-RKS iter  51:  -757.48994567344187   -3.97228e+00   1.88813e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  52:  -745.74970827989955    1.17402e+01   3.53212e-03 DIIS
   @DF-RKS iter  53:  -757.39021850958397   -1.16405e+01   1.95109e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  54:  -742.29363394537825    1.50966e+01   4.15438e-03 DIIS
   @DF-RKS iter  55:  -757.31429382078386   -1.50207e+01   1.98780e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  56:  -744.80037897342584    1.25139e+01   3.90143e-03 DIIS
   @DF-RKS iter  57:  -757.34136253519762   -1.25410e+01   1.97740e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  58:  -743.19611420959450    1.41452e+01   4.90019e-03 DIIS
   @DF-RKS iter  59:  -757.24424289556180   -1.40481e+01   2.04540e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  60:  -733.01391410551764    2.42303e+01   6.42992e-03 DIIS
   @DF-RKS iter  61:  -757.37349646367340   -2.43596e+01   1.94772e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  62:  -743.13690791546105    1.42366e+01   4.68034e-03 DIIS
   @DF-RKS iter  63:  -757.33796549339831   -1.42011e+01   1.99899e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  64:  -743.74115030604514    1.35968e+01   4.27803e-03 DIIS
   @DF-RKS iter  65:  -757.33320587591299   -1.35921e+01   2.00668e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  66:  -739.56682391361039    1.77664e+01   6.42121e-03 DIIS
   @DF-RKS iter  67:  -757.20223172407748   -1.76354e+01   2.06136e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  68:  -743.13882115637682    1.40634e+01   4.88563e-03 DIIS
   @DF-RKS iter  69:  -757.20419109086924   -1.40654e+01   2.06697e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  70:  -738.91777515340289    1.82864e+01   5.83716e-03 DIIS
   @DF-RKS iter  71:  -757.25848057559472   -1.83407e+01   2.01782e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  72:  -734.71055120204437    2.25479e+01   6.37064e-03 DIIS
   @DF-RKS iter  73:  -757.36941813327212   -2.26589e+01   1.97456e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  74:  -748.16578674330367    9.20363e+00   3.82829e-03 DIIS
   @DF-RKS iter  75:  -757.33699890417768   -9.17121e+00   1.99370e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  76:  -736.08445384495860    2.12525e+01   5.43510e-03 DIIS
   @DF-RKS iter  77:  -757.23850993244423   -2.11541e+01   2.03563e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  78:  -744.89642845964988    1.23421e+01   4.72037e-03 DIIS
   @DF-RKS iter  79:  -757.32955785791307   -1.24331e+01   1.98391e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  80:  -736.84461645727072    2.04849e+01   5.91030e-03 DIIS
   @DF-RKS iter  81:  -757.22294805052593   -2.03783e+01   2.05819e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  82:  -739.05843268532067    1.81645e+01   5.74478e-03 DIIS
   @DF-RKS iter  83:  -757.27958783135773   -1.82212e+01   2.01563e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  84:  -743.77591152812488    1.35037e+01   4.94459e-03 DIIS
   @DF-RKS iter  85:  -757.27638976122773   -1.35005e+01   2.03727e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  86:  -739.54365813714605    1.77327e+01   4.83445e-03 DIIS
   @DF-RKS iter  87:  -757.16782229193188   -1.76242e+01   2.07860e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  88:  -749.50408205693600    7.66374e+00   3.55313e-03 DIIS
   @DF-RKS iter  89:  -757.26126747239266   -7.75719e+00   2.00508e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  90:  -747.78893048091936    9.47234e+00   3.68546e-03 DIIS
   @DF-RKS iter  91:  -757.48004224340468   -9.69111e+00   1.91620e-03 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  92:  -734.14395956571877    2.33361e+01   6.17083e-03 DIIS
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  93:  -756.98088116826500   -2.28369e+01   2.20190e-03 DIIS
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  94:  -757.95743638517786   -9.76555e-01   1.78554e-03 DIIS
   @DF-RKS iter  95:  -759.85317643211795   -1.89574e+00   8.92026e-04 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  96:  -757.77254419159442    2.08063e+00   2.04014e-03 DIIS
   @DF-RKS iter  97:  -759.86786185265203   -2.09532e+00   8.71296e-04 SOKS, nmicro=40
Did not take a SOSCF step, using normal convergence methods
   @DF-RKS iter  98:  -756.63806624552649    3.22980e+00   2.28461e-03 DIIS
   @DF-RKS iter  99:  -759.79391928550524   -3.15585e+00   9.68819e-04 SOKS, nmicro=40
   @DF-RKS iter 100:  -758.45496753171210    1.33895e+00   1.31995e-03 SOKS, nmicro=40

PsiException: Could not converge SCF iterations in 100 iterations.

  Failed to converge.
Traceback (most recent call last):
  File "/apps/psi4conda/bin/psi4", line 333, in <module>
    exec(content)
  File "<string>", line 73, in <module>
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/driver.py", line 1339, in optimize
    G, wfn = gradient(lowername, return_wfn=True, molecule=moleculeclone, **kwargs)
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/driver.py", line 677, in gradient
    return driver_nbody.nbody_gufunc(gradient, name, ptype='gradient', **kwargs)
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/driver_nbody.py", line 291, in nbody_gufunc
    component_results = compute_nbody_components(func, method_string, metadata)
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/driver_nbody.py", line 502, in compute_nbody_components
    ptype_dict[pair], wfn = func(method_string, molecule=current_mol, return_wfn=True, **kwargs)
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/driver.py", line 739, in gradient
    wfn = procedures['gradient'][lowername](lowername, molecule=molecule, **kwargs)
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/procrouting/proc.py", line 2485, in run_scf_gradient
    ref_wfn = run_scf(name, **kwargs)
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/procrouting/proc.py", line 2390, in run_scf
    scf_wfn = scf_helper(name, post_scf=False, **kwargs)
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/procrouting/proc.py", line 1607, in scf_helper
    e_scf = scf_wfn.compute_energy()
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/procrouting/scf_proc/scf_iterator.py", line 93, in scf_compute_energy
    raise e
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/procrouting/scf_proc/scf_iterator.py", line 86, in scf_compute_energy
    self.iterations()
  File "/apps/psi4conda/lib//python3.8/site-packages/psi4/driver/procrouting/scf_proc/scf_iterator.py", line 473, in scf_iterate
    raise SCFConvergenceError("""SCF iterations""", self.iteration_, self, Ediff, Dnorm)

psi4.driver.p4util.exceptions.SCFConvergenceError: Could not converge SCF iterations in 100 iterations.


Printing out the relevant lines from the Psithon --> Python processed input file:
    core.set_global_option("MAXITER", 100)
    core.set_global_option("G_CONVERGENCE", "gau")
    core.set_global_option("SOSCF", "true")
    core.set_global_option("SOSCF_MAX_ITER", 40)
    set_num_threads(20)
--> E, wfn = optimize('B3LYP-D3', bsse_type='cp', return_wfn=True)
    fchk(wfn, 'citric_acid_bsse_cp.fchk')
    def saveFile(name, string):
        f=open(name, 'w')
        f.write(string)
        f.close()

!-------------------------------------------------------!
!                                                       !
!  Could not converge SCF iterations in 100 iterations. !
!                                                       !
!-------------------------------------------------------!

    Psi4 stopped on: Tuesday, 26 October 2021 02:03PM
    Psi4 wall time for execution: 3:28:16.19
```

**Comment by JonathonMisiewicz** [comment]
@BradenDKelly Any updates?

**Comment by BradenDKelly** [comment]
Hoping to know tomorrow. Personal issues had me out of office for longer than anticipated.

**Comment by BradenDKelly** [comment]
@JonathonMisiewicz It wasn't able to complete, it had a "bad alloc" error. However, the .dat file was 105,000 lines long, it had done many geometry iterations as well as scf iterations for each one, and the last scf iteration had completed in 8 steps. So it looked fine. My other jobs also ended prematurely, which is too bad. I have started them again. I think I need to add more memory to the machine...So I will make sure I don't overload the server this time. I am currently also running it using B3LYP/6-31G* just to see how that goes, since it is quick and dirty.

**Comment by BradenDKelly** [comment]
Well, the B3LYP/6-31G* had no issues with the SCF, it did hit 50 iterations and cancel because of that. It was clearly converging, albeit, very slowly. I am not sure why it stopped at 50 iterations considering I specify 100 iterations as my max?

For B3LYP/6-31G* I used

```
set {
    basis 6-31G*
    maxiter 100
    g_convergence gau
    opt_coordinates both
    }
```


```
 --------------------------------------------------------------------------------------------------------------- ~
   Step         Total Energy             Delta E       MAX Force       RMS Force        MAX Disp        RMS Disp  ~
  --------------------------------------------------------------------------------------------------------------- ~
      1   -1520.084027420038  -1520.084027420038      0.01209684      0.00318489      0.09585970      0.02642430  ~
      2   -1520.101971720320     -0.017944300282      0.00752027      0.00158941      0.12994949      0.02669655  ~
      3   -1520.111617832344     -0.009646112024      0.00329597      0.00094589      0.09913642      0.01723322  ~
      4   -1520.114648443302     -0.003030610957      0.00216198      0.00062847      0.08202824      0.01366308  ~
      5   -1520.116031406611     -0.001382963310      0.00122597      0.00033545      0.05989395      0.01111816  ~
      6   -1520.116772315849     -0.000740909238      0.00131990      0.00033374      0.06203108      0.01396358  ~
      7   -1520.117591979154     -0.000819663305      0.00119721      0.00034756      0.09267135      0.02072228  ~
      8   -1520.118547103847     -0.000955124693      0.00156371      0.00030629      0.07994782      0.01866175  ~
      9   -1520.119170359052     -0.000623255205      0.00125623      0.00027764      0.06344987      0.01578896  ~
     10   -1520.119638965232     -0.000468606180      0.00067629      0.00021266      0.04981669      0.01425496  ~
     11   -1520.119963148907     -0.000324183675      0.00106241      0.00018459      0.03721107      0.01011712  ~
     12   -1520.120138724090     -0.000175575183      0.00077654      0.00015862      0.03691754      0.00896792  ~
     13   -1520.120303579945     -0.000164855855      0.00055434      0.00012476      0.03805513      0.00913406  ~
     14   -1520.120449457803     -0.000145877858      0.00082349      0.00014532      0.03164180      0.00803309  ~
     15   -1520.120569511735     -0.000120053932      0.00090778      0.00014097      0.03673389      0.00918033  ~
     16   -1520.120702937426     -0.000133425691      0.00038130      0.00011705      0.03906407      0.00938723  ~
     17   -1520.120826640314     -0.000123702888      0.00049713      0.00013412      0.04869855      0.01017762  ~
     18   -1520.120963552205     -0.000136911891      0.00097786      0.00014542      0.06345799      0.01211031  ~
     19   -1520.121109731148     -0.000146178943      0.00076153      0.00011789      0.03636502      0.00661325  ~
     20   -1520.121197772284     -0.000088041137      0.00037865      0.00010028      0.03453021      0.00661186  ~
     21   -1520.121285576095     -0.000087803810      0.00058321      0.00010588      0.06658274      0.01288249  ~
     22   -1520.121382586266     -0.000097010172      0.00079341      0.00011526      0.04415119      0.00910505  ~
     23   -1520.121438772937     -0.000056186671      0.00038984      0.00008476      0.02879029      0.00675535  ~
     24   -1520.121480624119     -0.000041851182      0.00029135      0.00006923      0.02135187      0.00553512  ~
     25   -1520.121515817131     -0.000035193012      0.00048115      0.00007435      0.01822304      0.00479501  ~
     26   -1520.121550003861     -0.000034186730      0.00040549      0.00007623      0.02362958      0.00496087  ~
     27   -1520.121591602019     -0.000041598158      0.00042748      0.00007577      0.02764870      0.00495997  ~
     28   -1520.121636219730     -0.000044617711      0.00041419      0.00007626      0.01572951      0.00247943  ~
     29   -1520.121664199798     -0.000027980068      0.00036448      0.00005396      0.01558358      0.00247864  ~
     30   -1520.121687935156     -0.000023735358      0.00018694      0.00004151      0.04238212      0.00743914  ~
     31   -1520.121729400090     -0.000041464933      0.00025744      0.00005199      0.07801099      0.01549134  ~
     32   -1520.121775604285     -0.000046204195      0.00043952      0.00006717      0.03236441      0.00699519  ~
     33   -1520.121796595752     -0.000020991467      0.00038506      0.00005879      0.02493538      0.00652668  ~
     34   -1520.121823774411     -0.000027178659      0.00029803      0.00005011      0.02036796      0.00520408  ~
     35   -1520.121846068339     -0.000022293929      0.00024238      0.00005150      0.01478926      0.00385690  ~
     36   -1520.121862708408     -0.000016640069      0.00028916      0.00004749      0.02078889      0.00480948  ~
     37   -1520.121880934685     -0.000018226277      0.00021536      0.00004145      0.01307227      0.00313913  ~
     38   -1520.121891724924     -0.000010790239      0.00016879      0.00003664      0.01279721      0.00322554  ~
     39   -1520.121900131325     -0.000008406402      0.00013912      0.00003237      0.01438623      0.00370019  ~
     40   -1520.121908426307     -0.000008294981      0.00017824      0.00003228      0.01084486      0.00294248  ~
     41   -1520.121914889309     -0.000006463002      0.00017055      0.00003191      0.01109059      0.00280627  ~
     42   -1520.121921923314     -0.000007034006      0.00013705      0.00002926      0.01133335      0.00279490  ~
     43   -1520.121929534330     -0.000007611015      0.00014989      0.00003189      0.01190367      0.00275399  ~
     44   -1520.121936529658     -0.000006995328      0.00017664      0.00003210      0.01227925      0.00279053  ~
     45   -1520.121943471554     -0.000006941897      0.00014443      0.00002935      0.01310916      0.00279019  ~
     46   -1520.121949867482     -0.000006395927      0.00015012      0.00003049      0.01350550      0.00279008  ~
     47   -1520.121955858917     -0.000005991435      0.00021795      0.00003080      0.00611747      0.00139505  ~
     48   -1520.121960222608     -0.000004363691      0.00013063      0.00002234      0.02069125      0.00415818  ~
     49   -1520.121965013772     -0.000004791164      0.00010954      0.00002291      0.00896124      0.00182184  ~
     50   -1520.121967836686     -0.000002822914      0.00007375      0.00001974      0.01254476      0.00250384  ~
```

**Comment by BradenDKelly** [comment]
I don't get it, I ran another geom opt using B3LYP/6-31G* with the last coordinates of the previous attempt, and it converged 1st iteration

```
  **** Optimization is complete! (in 1 steps) ****

  ==> Optimization Summary <==

  Measures of convergence in internal coordinates in au.
  --------------------------------------------------------------------------------------------------------------- ~
   Step         Total Energy             Delta E       MAX Force       RMS Force        MAX Disp        RMS Disp  ~
  --------------------------------------------------------------------------------------------------------------- ~
      1   -1520.121971894730  -1520.121971894730      0.00006927      0.00001834      0.00081821      0.00021391  ~
  --------------------------------------------------------------------------------------------------------------- ~
```

**Comment by JonathonMisiewicz** [comment]
This issue was originally about issues with converging an SCF, and now I'm not sure what it's about. I need you to re-focus the issue.

Are you saying that for the same geometry, Psi is giving different energies/gradients depending on which optimization run it was part of?

**Comment by hokru** [comment]
Please check the output for additional information or printouts from the optimizer for the 1-step optimization, or upload the file.

**Comment by BradenDKelly** [comment]
I will stick to the SCF issue and report the results once the B3LYP-D3/def2-tzvp geom opt is done. 

I was noting that it seemed odd that one optimization did 50 iterations and petered out, but upon restarting, it converged in one step. Different wavefunctions though, the SCF in the "restart" started from SAD guess, the 50th iteration of the first try was using previous orbitals as initial guesses, I turned off permanent SAD guess for it.

In short, my apologies for muddying the water, I think just ignore, and I will report the results for the initial optimization problem once they finish. I have both `opt_coordinates both` as well as `opt_coordinates both` and `guess sad`
    `guess_persist true` running.

**Comment by BradenDKelly** [comment]
So, the dimer converged in 57 geometry iterations using
```
set {
    basis def2-tzvp
    maxiter 100
    geom_maxiter 200
    g_convergence gau
    opt_coordinates both
    guess sad
    guess_persist true
    }
```
In a different run, it hit the maximum geometry iterations of 50, but was running fine using
```
set {
    basis def2-tzvp
    maxiter 100
    g_convergence gau
    soscf true
    soscf_max_iter 40
    guess sad
    guess_persist true
    opt_coordinates both
    }
So, there is no need for `soscf true` and `soscf_max_iter`
```
Same story when using
```
set {
    basis def2-tzvp
    maxiter 100
    g_convergence gau
    opt_coordinates both
    }
```
here the max iterations of 50 geometry iterations was hit, but it was clearly converging.

**Comment by JonathonMisiewicz** [comment]
Are we still talking about SCF iterations not converging? It sounds like there was never an SCF convergence problem for this molecule.

This raises troubling questions about why any of these input files converge in different number of iterations: energies and gradients and therefore geometry steps should be identical but for numerical noise... However, that would be for a separate topic, if reproducible.

**Comment by BradenDKelly** [comment]
Well, initially it was believed to be SCF related because that was the error at program failure. However, it now seems that the reason for the SCF failing was caused by the geometry iteration producing unphysical geometries, which then led to the next SCF iteration failing. By fixing the geometry iteration, and it appears that 'opt_coordinates both' does the trick, no unphysical geometries occur, and the program purs along to completion. In the case of these dimers, completion takes more than the default 50 geometry iterations. Above, one succeeded in 57 steps, two stopped at 50 steps but were clearly converging, they just hit their geom_maxiter of 50 steps and stopped.

**Comment by JonathonMisiewicz** [comment]
Okay, so what needs to be done to close the issue?

**Comment by BradenDKelly** [comment]
It may be closed.

Solution appears to be to include "opt_coordinates both"
